module/zfs/lz4.c

```diff
@@ -1014,6 +1014,23 @@ LZ4_uncompress_unknownOutputSize(const char *source, char *dest, int isize,
 	return (-1);
 }
 
+#ifdef __FreeBSD__
+/*
+ * FreeBSD has 4, 8 and 16 KB malloc zones which can be used here.
+ * Should struct refTables get resized this may need to be revisited, hence
+ * compile-time asserts.
+ */
+_Static_assert(sizeof (struct refTables) <= 16384,
+	"refTables too big for malloc");
+_Static_assert((sizeof (struct refTables) % 4096) == 0,
+	"refTables not a multiple of page size");
```

Comment:
IIRC FreeBSD has power-of-two zones from 64B to 64KB.  So above checks are not exactly correct.

Reply:
It is stricter than necessary, but it does not make it incorrect. Should this grow, someone should revisit what makes the most sense.